### ISSUE TITLE
Touch-bar volume control disabled

### ISSUE THREAD

**Issue by ruzz311** [romankisil/eqMac2#50]
The touch bar on my 2016 macbook disabled when EQ2 is running.  It's not a new bug.  I've experienced this since I've installed 2.0.5. I've kept eq up-to-date and also saw it in 2.0.6 and pre-release of 2.0.7.  

![eq2-touchbar-disabled](https://user-images.githubusercontent.com/419608/30217449-c928eea8-947b-11e7-86df-2a2425d88d92.jpg)

**Comment by romankisil** [comment]
This was reported many times and obviously I am aware of this. It will be fixed once we switch to a different routing mechanism.

**Comment by romankisil** [comment]
This will be fixed once #51 is done

**Comment by romankisil** [comment]
Moving to a visible audio device in release of build 2.1
Fixed by issue #62

**Comment by Bigtuna00** [comment]
This should be reopened since #62 is no longer fixed (and will stop people like me from almost filing a duplicate :)

**Comment by romankisil** [comment]
@Bigtuna00 agreed

**Comment by easdale** [comment]
I have the same issue

**Comment by romankisil** [comment]
Build 2.1 has just been released. You can download it from https://bitgapp.com/eqmac
Could you please update and report if the issue is still relevant. Thanks :)

**Comment by easdale** [comment]
Hi
Touch Bar now works and so much easier having everything open in the one window from the top bar.
Its weird though how if you click on the icon in the top of my screen the panel opens as normal but it won’t close when I click on the icon.
The only way to get rid of the emac window is to do like a command/tab and open another application to cancel the opt up of eqmqc at the top.
Every other icon on the top of my screen will close if I click on the I can for a second time.
Thats for updating the Touch Bar glitch so quickly.
Adrian

**Comment by Bigtuna00** [comment]
Yes touch bar issues seem to be fixed. Agree with @easdale about the clicking behavior, it's not "normal" but that should be filed as a separate issue.

**Comment by easdale** [comment]
Theres also another issue. I have no bass any more which I noticed on the last version as well. When I use a flat profile with no eq the internal speakers sound very tinny.

**Comment by Bigtuna00** [comment]
Spoke too soon. Touch bar responds to touch, this is true, but adjusting the volume does nothing.  My volume is at 100% no matter where the volume slider is.  Luckily the mute function works.

**Comment by easdale** [comment]
Touch Bar responds on my MacBook Pro

**Comment by romankisil** [comment]
@Bigtuna00 what audio device are you trying to control the volume of?

**Comment by easdale** [comment]
I am referring to the sound of the speakers built into my Touch Bar 15” MacBook Pro

**Comment by Bigtuna00** [comment]
@romankisil this was with "Internal Speakers".

**Comment by romankisil** [comment]
My Touchbar controls the internal speakers just fine :( I always suggest people restart their Mac so solve many Sound related issues.

**Comment by Bigtuna00** [comment]
I did restart as a troubleshooting step prior to posting. Sorry, should have mentioned that. I will try clean installing again. The uninstall from the previous version doesn't appear to have worked. What can/should I delete to clean it all out? Any logs or anything I should attach? Thanks.

**Comment by romankisil** [comment]
when you install the new build I make sure to clear everything from the old builds, so nothing to delete anymore really

**Comment by Bigtuna00** [comment]
Do you mean just delete the `.app`?

**Comment by romankisil** [comment]
Well, to delete the app it would be recommended to use the Uninstall button within the app

**Comment by Bigtuna00** [comment]
Here's what I tried:

* Installed the lastest download
* (volume max'd no matter what)
* Ran the uninstall
* Rebooted
* Installed the lastest download
* Rebooted

Volume is still max no matter what.

I did note that it used to say "aggregate device" or some such under System Prefs whereas now it doesn't list a `Type` at all.

**Comment by psxcode** [comment]
@Bigtuna00 Your case could relate to #103

**Comment by Bigtuna00** [comment]
That's a good point. Sorry @romankisil I should have mentioned I was only testing system sounds, I haven't even tried anything else yet.

**Comment by nodeful** [comment]
Hey everyone. Excuse me for the lack of updates. I have been really busy on the new version of eqMac. I am happy to announce an open beta release. 
You can download it here: https://eqmac.app
There are a lot of known bugs that I could not fix just yet:

- Driver not installing from the first try. Sometimes need to restart eqMac a couple times.
- Complex Device switching logic is not perfect (for example when switching off the device that's currently playing, plugging in new device etc.)
- UI takes long to load (a lot can be done to improve that)
- UI dimensions are not synching properly
- There's a noticeable audio delay in the playback, so not suitable for video yet (I know where the problem lies just haven't had the time to fix yet)
- Sometimes the playback doesn't start, in that case please try to switch the output to the device you want and eqMac should try and build the audio pipeline again.

So as you can see there's still some work to be done but at least it's a completely modern tech stack that I am fully willing to work on: 100% Swift, User Space driver and a Web UI, instead of C++, Kernel Drivers. As I said, should work on Catalina and should not have any pesky ear drum blasting volume bugs (at least I didn't come across any).
I'm open to feedback and issue reports here: https://github.com/bitgapp/eqmac
That repo will become the main repo for eqMac and this repo will be archived eventually.
And yes it's called just eqMac, not eqMac3 :) And it will stay that way forever.

**Comment by nodeful** [comment]
Hey everyone 👋 
I will be open sourcing the new version of eqMac, therefore, I am closing this issue and repository as I will be transfering [nodeful/eqMac2](https://github.com/nodeful/eqMac2) repo to [bitgapp/eqMac](https://github.com/bitgapp/eqMac)
Please bear with me for the next couple of days as I setup that new repo. The eqMac2 code will be living in the [old-eqMac2](https://github.com/bitgapp/eqMac/tree/old-eqmac2) branch if anyone is interested in referencing it for whatever reason.
Please try out the new version and report bugs to the new repo.
Thanks for using eqMac.
